src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -114,6 +128,27 @@ def put(self, request: Request, organization, project, monitor) -> Response:
             params["status"] = result["status"]
         if "config" in result:
             params["config"] = result["config"]
+
+            # update timeouts + expected next check-in, as appropriate
+            checkin_margin = result["config"].get("checkin_margin")
+            if checkin_margin != existing_margin:
+                for monitor_environment in MonitorEnvironment.objects.filter(monitor_id=monitor.id):
+                    monitor_environment.next_checkin_latest = (
+                        monitor_environment.next_checkin
+                        + timedelta(minutes=int(checkin_margin or 1))
+                    )
+                    monitor_environment.save(update_fields=["next_checkin_latest"])
```

Comment:
Instead of a for loop I think you should be able to write this as an `UPDATE` query that does the date math in postgres itself.

What do you think @dfuller

Reply:
Yeah doing that can be a good idea to avoid concurrency bugs. You can use `F()` to do that kind of thing, we do it to update counts usually

Reply:
struggling with how to update a `DateTimeField` using `.replace()` or something similar. Django wants to do this at the SQL level or something. maybe I can call another helper function? will PM you

Reply:
nvm got it